### PR TITLE
Feature: ApolloClient v3

### PR DESCRIPTION
**Pull request by afjlambert**
Please squash and merge.

### CODE REVIEW

File: apps/client/src/apolloClient.js [hunk 1]
```diff
@@ -1,47 +1,6 @@
-import { MockLink } from "@apollo/react-testing";
-import {
-  InMemoryCache,
-  IntrospectionFragmentMatcher,
-} from "apollo-cache-inmemory";
-import { ApolloClient } from "apollo-client";
-import { ApolloLink } from "apollo-link";
-import { onError } from "apollo-link-error";
-import { HttpLink } from "apollo-link-http";
+import { ApolloClient, InMemoryCache } from "@apollo/client";
```
Comment (robinpiets) [inline apps/client/src/apolloClient.js:1]: WOOW!

File: apps/client/src/utils/test-utils.js [hunk 2]
```diff
@@ -23,6 +23,12 @@ const AllTheProviders = ({ children, mocks }) => (
   </SessionProvider>
 );
 
+export const getTestClient = (mocks) =>
+  new ApolloClient({
+    link: new MockLink(mocks, true),
+    cache: new InMemoryCache(),
+  });
+
```
Comment (robinpiets) [inline apps/client/src/utils/test-utils.js:31]: 🥇

File: apps/client/sttr_build/parser.ts [hunk 3]
```diff
@@ -0,0 +1,304 @@
+import { createHash } from "https://deno.land/std/hash/mod.ts";
```
Comment (robinpiets) [inline apps/client/sttr_build/parser.ts]: Why is `parser.ts` with deno in this PR?

**Review by robinpiets — CHANGES_REQUESTED** [review]
## 1: I get this message when starting the dev server in terminal:
```
npm WARN graphql-tag@2.10.3 requires a peer of graphql@^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none is installed. You must install peer dependencies yourself.
```

## 2: I get this warning in console:
```
index.js:1 Warning: Cannot update a component (`SessionProvider`) while rendering a different component (`Unknown`). To locate the bad setState() call inside `Unknown`, follow the stack trace as described in https://fb.me/setstate-in-render
    in Unknown (created by Context.Consumer)
    in Route (at Router.js:20)
    in Switch (at Router.js:13)
    in Suspense (at Router.js:12)
    in Router (created by BrowserRouter)
    in BrowserRouter (at Router.js:10)
    in Router (at src/index.js:44)
    in MatomoProvider (at src/index.js:43)
    in ThemeProvider (created by ThemeProvider)
    in ThemeProvider (at src/index.js:40)
    in ApolloProvider (at src/index.js:39)
    in SessionProvider (at src/index.js:38)
```

## 3: Why is parser.ts with deno in this PR?


Everything else seems to work though 😃

File: apps/client/sttr_build/parser.ts (hunk 3 above)
Comment (afjlambert) [reply apps/client/sttr_build/parser.ts]: huh?

**Comment by afjlambert** [comment]
@robinpiets 
I cannot reproduce `2`. The rest is fixed. Can you check again?
Please provide screenshot / reproduce steps if the problem stil exists.

**Comment by robinpiets** [comment]
@afjlambert 
I still get his error:



Steps to reproduce:
Open new browser window, open console, open a random checker: `http://localhost:3000/test-outcomes` and you see the error. (This only seems happens on the first init of a checker. After reload you cannot reproduce other then opening a new checker)

**Comment by robinpiets** [comment]
Update: this warning is also in `chappie2.0` branch. Will make a new ticket for this

**Review by robinpiets — APPROVED** [review]

**Review by svenjens — APPROVED** [review]

**Pull request merged**